### PR TITLE
[Toast] Updates to fit a more compact layout

### LINKED ISSUE #7766: [Toast] Update styles to a more compact layout
We want to update the toast to a more compact layout to align with the density adjustments being made in the admin and to the Polaris design system. [See Figma](https://www.figma.com/file/wgbt21f1PZm5lrUecbB5oY/%F0%9F%8D%9E-Toast-component-rework?t=t1DGV6KQXGXBfkaQ-0).

For example:

Screenshot 2022-11-18 at 11 41 40 AM

### Font
Desktop
- [ ] Update font size to 14px
- [ ] Update font weight to 500 (medium)
- [ ] Update line height to 20px (if it’s not implicit already)

Mobile
- [ ] Update font size to 15px (including the undo link)
- [ ] Font weight is still 500 (medium)

### Padding
Desktop

Screenshot 2022-11-18 at 11 43 19 AM

- [ ] Also need to reduce the padding between the content, action, and close button

Mobile

Screenshot 2022-11-18 at 11 43 21 AM

- [ ] Also need to reduce the padding between the content, action, and close button

#### Border radius
- [ ] Update border radius to `4px`

### Color
- [ ] <del>Update the undo link from black to gray, unless the type is an error (--p-border-hovered)</del>
- [ ] <del>Update the close button from white to gray, unless the type is an error (--p-border-hovered)</del>
Update: We shouldn't use border colors to update text.

### Size
- [ ] The close button’s width/height is decreased to 10px

### Icon
- [ ] An error icon is added when the type is an error

cc @jameswfindlater 

More details [here](https://docs.google.com/document/d/1OAzSJaz4-bH21Lr9DBt1cY_8uGFrzPOYdqc5b-u2xEg/edit#).

### PR DESCRIPTION
**Pull request by n-filatov**
### WHY are these changes introduced?

Resolves: https://github.com/Shopify/polaris/issues/7766

Updates the toast to fit a more compact layout. 

Figma: https://www.figma.com/file/wgbt21f1PZm5lrUecbB5oY/%F0%9F%8D%9E-Toast-component-rework?t=t1DGV6KQXGXBfkaQ-0

### WHAT is this pull request doing?

Redesign screenshots:

<details>
<summary>Default</summary>

Before:
default Toast component before changes

After:
default Toast component after changes

</details>

<details>
<summary>With Action</summary>

Before:
Toast with action before changes

After:
Toast with action after changes
</details>

<details>
<summary>Error</summary>

Before:
Toast with error before changes

After:
Toast with error after changes
</details>

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

#### Test in spin

Spin instance: https://admin.web.nf-tost-web.nikita-filatov.eu.spin.dev/store/shop1/products?selectedView=all

<details>
<summary>Default toast</summary>

Create a new product here: [link](https://admin.web.nf-tost-web.nikita-filatov.eu.spin.dev/store/shop1/products/new)

**Expected result**
image
</details>

<details>
<summary>Toast with error</summary>

1. Create a new product here: [link](https://admin.web.nf-tost-web.nikita-filatov.eu.spin.dev/store/shop1/products/new)

2. Before clicking on the Save button disable the wifi:
image

**Expected result:**
image
</details>

#### Test on local machine

1. Open http://localhost:6006/?path=/story/all-components-toast--default
5. Check
2.1 Default toast
2.2. Toast with action
2.3 Toast with error

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by sarahill — APPROVED** [review]
This looks great!

File: .changeset/perfect-games-cough.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': major
+---
+
+Redesign the Toast component to a more compact layout
```
Comment (kyledurand) [inline .changeset/perfect-games-cough.md]:
```suggestion
Updated the Toast component to a more compact layout
```

File: polaris-react/src/components/Frame/components/Toast/Toast.scss [hunk 2]
```diff
@@ -21,41 +21,48 @@ $Backdrop-opacity: 0.88;
...
   background: var(--p-action-critical);
   color: var(--p-text-on-critical);
 
   .CloseButton {
-    fill: var(--p-icon-on-critical);
+    color: var(--p-icon-on-critical);
   }
 }
 
+.LeftIcon {
```
Comment (kyledurand) [inline polaris-react/src/components/Frame/components/Toast/Toast.scss]: Can we call this `LeadingIcon` or just `Icon` instead? We're working towards removing left / right / top / bottom phrasing from the system. Not a blocker but if you could make the change here and to `leftIconMarkup` => `leadingIconMarkup` or `iconMarkup` that would be awesome

**Review by kyledurand — APPROVED** [review]
Looks great to me 👍

**Comment by heyjoethomas** [comment]
Can we double check the tokens being used here? In the Figma doc there are lots of incorrect tokens being used. For example, we shouldn't be using border tokens to style text.

File: polaris-react/src/components/Frame/components/Toast/Toast.scss (hunk 2 above)
Comment (n-filatov) [reply polaris-react/src/components/Frame/components/Toast/Toast.scss]: Got it, thank you for feedback 🙏

Fixed:
https://github.com/Shopify/polaris/blob/173bbf95a89aa93bde219f0c0566f381fc07be72/polaris-react/src/components/Frame/components/Toast/Toast.tsx#L65-L69

File: .changeset/perfect-games-cough.md (hunk 1 above)
Comment (n-filatov) [reply .changeset/perfect-games-cough.md]: Thanks, fixed!

**Comment by n-filatov** [comment]
> Can we double check the tokens being used here? In the Figma doc there are lots of incorrect tokens being used. For example, we shouldn't be using border tokens to style text.

Thanks for the very useful comment, Joe! 
@jameswfindlater is now on vacation till the next week. I will meet with him to clarify the tokens which we should use here. Until then, I will mark the PR WIP

**Comment by heyjoethomas** [comment]
> > Can we double check the tokens being used here? In the Figma doc there are lots of incorrect tokens being used. For example, we shouldn't be using border tokens to style text.
> 
> Thanks for the very useful comment, Joe! @jameswfindlater is now on vacation till the next week. I will meet with him to clarify the tokens which we should use here. Until then, I will mark the PR WIP

Sounds good. If y'all want to jam together on it just reach out. Happy to jump in a session and collab on it.

**Comment by sarahill** [comment]
Thanks for catching that @heyjoethomas. I noticed that when I was adding them to the UI Kit. Here's a link to the [branch in Figma](https://www.figma.com/file/4dAAt5iFPSaxUKiYVKrkYj/branch/BQX9uuadoVUFTcCfJ2lY2Z/Polaris-Components?node-id=364%3A1236&t=JjVXM0AtK92Sc5E2-1)

**Comment by n-filatov** [comment]
> Thanks for catching that @heyjoethomas. I noticed that when I was adding them to the UI Kit. Here's a link to the [branch in Figma](https://www.figma.com/file/4dAAt5iFPSaxUKiYVKrkYj/branch/BQX9uuadoVUFTcCfJ2lY2Z/Polaris-Components?node-id=364%3A1236&t=JjVXM0AtK92Sc5E2-1)

@sarahill thanks for the Figma link. I have changed the tokens in my PR. The main UI difference is the color for action text, it will be like the main text(using --p-text-on-interactive token)

https://github.com/Shopify/polaris/blob/e8d063dabfa257df3dbe91523c093cbc83c5693e/polaris-react/src/components/Frame/components/Toast/Toast.scss#L25

https://github.com/Shopify/polaris/blob/e8d063dabfa257df3dbe91523c093cbc83c5693e/polaris-react/src/components/Frame/components/Toast/Toast.scss#L51

It will look like this:

Tooltip text on interactive

**Review by sarahill — APPROVED** [review]
Thanks for updating the tokens. Looks good 👍

**Review by LauraAubin — APPROVED** [review]
Great work! Tophat looks good

Screenshot 2022-11-23 at 3 46 44 PM

Screenshot 2022-11-23 at 12 20 51 PM

Screenshot 2022-11-23 at 12 20 28 PM

**Comment by LauraAubin** [comment]
I think we should also create an issue to update the action text/close button color when there are available tokens.

**Review by sarahill — CHANGES_REQUESTED** [review]
Tokens are good but I just noticed 2 things:
- Close button isn't center aligned when there's an action present
- The designs look like they have a shadow but I'm not seeing that here.. correct me if I'm missing it though: `shadow-popover`

close button not center
image

shadow (it's subtle)
image

cc: @jameswfindlater

**Comment by n-filatov** [comment]
> Tokens are good but I just noticed 2 things:
> 
> * Close button isn't center aligned when there's an action present
> * The designs look like they have a shadow but I'm not seeing that here.. correct me if I'm missing it though: `shadow-popover`
> 
> close button not center image
> 
> shadow (it's subtle) image
> 
> cc: @jameswfindlater

Yea, my bad. Thanks for noticing 🙌.

1. Added box-shadow:
https://github.com/Shopify/polaris/blob/9cd736016083755a970399e440f7771a79377579/polaris-react/src/components/Frame/components/Toast/Toast.scss#L13

2. Fixed cross icon position:
image

**Review by sarahill — APPROVED** [review]

**Review by LauraAubin — APPROVED** [review]

**Comment by sarahill** [comment]
Following up on UI Kit updates here: I went ahead and merged in the `Toast` updates to the UI Kit and will publish those changes to the library one this branch is merged 😄

**Review by jameswfindlater — APPROVED** [review]
Looking great ... Thank you everyone for doing this!

**Pull request merged**